File: android/multando-sdk/src/main/kotlin/com/multando/sdk/services/ReportService.kt
package com.multando.sdk.services

import com.multando.sdk.core.HttpClient
import com.multando.sdk.models.*

/**
 * CRUD operations on traffic violation reports.
 */
class ReportService internal constructor(
    private val httpClient: HttpClient
) {

    /** Create a new report. */
    suspend fun create(report: ReportCreate): ReportDetail =
        httpClient.request(
            method = "POST",
            path = "/api/v1/reports",
            body = report
        )

    /** Fetch a paginated list of reports. */
    suspend fun list(
        page: Int = 1,
        pageSize: Int = 20,
        status: ReportStatus? = null
    ): ReportList {
        val params = mutableMapOf(
            "page" to page.toString(),
            "page_size" to pageSize.toString()
        )
        status?.let { params["status"] = it.name.lowercase() }

        return httpClient.request(
            method = "GET",
            path = "/api/v1/reports",
            queryParams = params
        )
    }

    /** Fetch a single report by ID. */
    suspend fun get(id: String): ReportDetail =
        httpClient.request(method = "GET", path = "/api/v1/reports/$id")

    /** Update an existing report. */
    suspend fun update(id: String, report: ReportCreate): ReportDetail =
        httpClient.request(
            method = "PUT",
            path = "/api/v1/reports/$id",
            body = report
        )

    /** Delete a report. */
    suspend fun delete(id: String) {
        httpClient.requestVoid(method = "DELETE", path = "/api/v1/reports/$id")
    }
}

File: ios/Sources/MultandoSDK/Services/ReportService.swift
import Foundation

/// CRUD operations on traffic violation reports.
public final class ReportService: Sendable {

    private let httpClient: HTTPClient

    init(httpClient: HTTPClient) {
        self.httpClient = httpClient
    }

    /// Create a new report.
    public func create(_ report: ReportCreate) async throws -> ReportDetail {
        try await httpClient.request(
            method: "POST",
            path: "/api/v1/reports",
            body: report
        )
    }

    /// Fetch a paginated list of reports.
    public func list(page: Int = 1, pageSize: Int = 20, status: ReportStatus? = nil) async throws -> ReportList {
        var queryItems = [
            URLQueryItem(name: "page", value: String(page)),
            URLQueryItem(name: "page_size", value: String(pageSize))
        ]
        if let status {
            queryItems.append(URLQueryItem(name: "status", value: status.rawValue))
        }
        return try await httpClient.request(
            method: "GET",
            path: "/api/v1/reports",
            queryItems: queryItems
        )
    }

    /// Fetch a single report by ID.
    public func get(id: String) async throws -> ReportDetail {
        try await httpClient.request(
            method: "GET",
            path: "/api/v1/reports/\(id)"
        )
    }

    /// Update an existing report.
    public func update(id: String, report: ReportCreate) async throws -> ReportDetail {
        try await httpClient.request(
            method: "PUT",
            path: "/api/v1/reports/\(id)",
            body: report
        )
    }

    /// Delete a report.
    public func delete(id: String) async throws {
        try await httpClient.requestVoid(
            method: "DELETE",
            path: "/api/v1/reports/\(id)"
        )
    }
}
